### DISCUSSION TITLE
o1 cannot response when it think for a long time

### DISCUSSION THREAD

**Discussion by SaltedFish23** [open-webui/open-webui#6127] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.32

- **Operating System:**Windows 11, WSL2

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [ ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

I can see the response.

## Actual Behavior:

![image](https://github.com/user-attachments/assets/cee3d4f8-93ea-4b55-868a-e28c3e06dc3e)

![image](https://github.com/user-attachments/assets/78c60817-6b13-4342-80e5-bc60b8337440)
The log say that i got a timeout.

## Description

**Bug Summary:**
If you ask a long or hard question, o1-mini may have to think for a long time. And you will get an error of timeout while the api work normally.
![image](https://github.com/user-attachments/assets/2efdecc0-65b3-460b-803a-c81e76f5090a)
Api says that it need 84s, so i got a timeout.

## Reproduction Details

**Steps to Reproduce:**
- Answer o1 a quetion need a long time to response.

## Logs and Screenshots

**Docker Container Logs:**
```
"POST /api/chat/completions HTTP/1.1" 504 Gateway Timeout
```

**Comment by yuzukumo** [comment]
Same problem. I find that there exists an environment variable AIOHTTP_CLIENT_TIMEOUT, but I'm not sure if it refers to this timeout settings.

**Reply by SaltedFish23** [reply]
Have you use API redistribution systems like one-api, new-api or something like that? I find that my new-api send 504 to Open webui (it did not have record on that 84s-o1-mini). So maybe i have to adjust my new-api :( although i don't know what should i do.

**Reply by yuzukumo** [reply]
> Have you use API redistribution systems like one-api, new-api or something like that? I find that my new-api send 504 to Open webui (it did not have record on that 84s-o1-mini). So maybe i have to adjust my new-api :( although i don't know what should i do.

yes. i use one hub to forward requests. i haven't found a way to adjust this timeout variable in one hub. and no record as well.

**Reply by yuzukumo** [reply]
> Have you use API redistribution systems like one-api, new-api or something like that? I find that my new-api send 504 to Open webui (it did not have record on that 84s-o1-mini). So maybe i have to adjust my new-api :( although i don't know what should i do.

after update of the api redistribution, now the record in api redistribution is right. but openwebui still occurs timeout error

**Reply by SaltedFish23** [reply]
I did nothing but it runs well, and my open webui can receive 87s-o1-mini now, while i can see the record in api redistribution.
I have done these thing:
- Disable pipeline on open webui
